overlord/snapstate/policy/os.go

```diff
@@ -56,6 +61,26 @@ func (p *osPolicy) CanRemove(st *state.State, snapst *snapstate.SnapState, rev s
 		return errIsModel
 	}
 
+	if rev.Unset() || len(snapst.Sequence.Revisions) == 1 {
```

Comment:
I wouldn't add the 2nd check, rev Unset is used here in other places under the same assumptions. What I would do instead is, make sure that various test fails if we change the code in snapstate.go. Improve the doc comment in Policy to clarify the point that Rev should be pass unset in all situation where all the revisions go away independently whether the user passed a revision or not